### ISSUE TITLE
Development workflow

### ISSUE THREAD

**Issue by Wykks** [dherges/ng-packagr#78]
Hi !
I'm not seeing a way to watch files changes with ng-packagr, do you plan to add this ?
Edit: using `ng-packaged` if I run `yarn build:lib` while `ng serve` is running, it break.
How do you develop a lib with ng-packagr ? (sorry I'm totally lost...)

**Comment by dherges** [comment]
@Wykks When using together with Angular CLI, right now, I let the CLI re-compile the build, i.e. `yarn build:lib && ng serve <control>+<c>`. Angular CLI sometimes doesn't pick up file changes and then needs to be restarted.

For watching file changes, I use an separate `karma.conf.js` which is running tests for a lib. So I do get "watch mode" from running the test suite.

**Comment by Wykks** [comment]
Okay, I see, thanks !
I think I'll stay with the starter of filipesilva for the time being since watch mode works fine there. Unit test in watch mode are not enough for me.

**Comment by dherges** [comment]
@Wykks One more note: it is possible to use ng-packagr together with a generic file watcher tool like [chokidar](https://www.npmjs.com/package/chokidar#getting-started) or [chokidar-cli](https://github.com/kimmobrunfeldt/chokidar-cli#usage).

From their docs, you can try:

```bash
$ chokidar '**/*.html' '**/*.ts' | ng-packagr
```

**Comment by ronnyek** [comment]
So I tried this with ng-serve on the test app, and then chokidar for the library build itself, but the ng serving of the dev app basically catches the library disappear, doesn't catch the complete of the rebuild.

Has anyone got this process polished enough where they can tweak ts/html for components in the library and rebuild automatically, all the while still watching the demo app and having that "just work"?

**Comment by Tamango92** [comment]
@ronnyek , Hello.
I've built simple library example, which is based on basic angular-cli app and ng-packagr. It has preview app inside, supports demo/developer mode and can be easily imported into your running angular5 apps. I've also added compodoc and live-rebuild mode to make it a bit more cool.

Check it out here: https://github.com/Tamango92/yo-lib
Read the docs (sorry in advance, I am not good in writing nice docs), feel free to rename lib in local package.json file and replace all 'yo-' prefixes with your own.
